rt_entt_codegen/shared/parallel.cc

```diff
@@ -39,28 +39,31 @@ static auto is_capability_safe(ecsact_system_capability capability) -> bool {
 }
 
 static auto loop_iterator(
-	const std::vector<ecsact_system_like_id>&          system_list,
-	const std::vector<ecsact_system_like_id>::iterator begin,
-	std::vector<std::vector<ecsact_system_like_id>>&   parallel_system_cluster
+	const std::vector<system_like_id_variant>&          system_list,
+	const std::vector<system_like_id_variant>::iterator begin,
+	std::vector<std::vector<system_like_id_variant>>&   parallel_system_cluster
 ) -> void {
-	std::vector<ecsact_system_like_id> parallel_system_list;
+	std::vector<system_like_id_variant> parallel_system_list;
 	auto unsafe_comps = std::set<ecsact_component_like_id>{};
 
 	using ecsact::meta::decl_full_name;
 
 	for(auto iterator = begin; iterator != system_list.end(); iterator++) {
-		auto sys_like_id = *iterator;
-		auto capabilities = ecsact::meta::system_capabilities(sys_like_id);
+		auto sys_like_id_variant = *iterator;
+		auto capabilities =
+			ecsact::meta::system_capabilities(sys_like_id_variant.get_sys_like_id());
 
-		auto generate_ids = ecsact::meta::get_system_generates_ids(sys_like_id);
+		auto generate_ids = ecsact::meta::get_system_generates_ids(
+			sys_like_id_variant.get_sys_like_id()
```

Comment:
we should add an implicit cast. There's a bunch of these calls that couldbe simplified.
```suggestion
			sys_like_id_variant
```